### PR TITLE
Document new ways to add component/variant notes

### LINKED ISSUE #38: Update component readme docs
Docs need updating following merge of https://github.com/frctl/fractal/pull/566

### PR DESCRIPTION
**Pull request by rchavik**
Closes #38

### CODE REVIEW

File: docs/guide/components/notes.md [hunk 1]
```diff
@@ -10,13 +10,18 @@ Whilst Fractal lets you create rich standalone [documentation pages](../document
 
 ## The README.md file
 
-[Compound components](../components/#compound-components) can include a `README.md` file within their component folder:
+[Components](../components/#components) and [variants](../component/#variants) can include notes in a `README.md`,
+`<component-name>.readme.md`, or `<component-name>--<variant-name>.readme.md` file within their component folder:
```
Comment (LeBenLeBen) [inline docs/guide/components/notes.md:14]: It's a bit weird to have it in uppercase once and lowercase on the others. Should we specify that it's case insensitive?

Comment (risker) [inline docs/guide/components/notes.md:14]: it's a bit messy here. with the new logic, we allow readme files for non-compound components as well, so "within their component folder" is not 100% correct. maybe "beside the component template file" would be better?

for compound components, we allow the main readme file to be added without the component name (similarly to how config files work), so just `readme.md` works as well as `component.readme.md`. this should stay documented as well.

**Comment by rchavik** [comment]
Should we recomment to adopt <component>.readme.md (instead of just readme.md) ?

**Comment by risker** [comment]
I wouldn't recommend one over the other, in the end it's the users choice.

I think this is fine now, but let's wait with the merging until core is released.

**Pull request merged**